### DISCUSSION TITLE
Mode jumping adaptive MCMC

### DISCUSSION THREAD

**Discussion by ben18785** [pints-team/pints#1563] [New methods]
[This](https://pdfs.semanticscholar.org/c75d/f035c23e3c0425409e70d457cd43b174076f.pdf) note by Chris Holmes is interesting and suggests a way to sample from multimodal posteriors if the location of the modes is known. I wonder whether a combination of particle swarm/other optimiser which actually gets stuck on the modes could be used to find the modes, then Holmes' algorithm to sample from them.

**Comment by ben18785** [comment]
@MichaelClerx @martinjrobins Is there a way we could adapt PSO/CMAES etc. to allow it to return all the modes it finds in a distribution? Alternatively, are there algorithms that do this?

**Comment by martinjrobins** [comment]
I doubt many global optimisation algorithms would be able to find alternate modes, as their problem statement is to specifically avoid local optimum in favour of the single global optimum. I'm sure that there are plenty of algorithms to do this, the simplest one I can think of is a multi-start optimisation (just start multiple local optimisation algorithms, eg. gradient descent, at random locations in the parameter space)

**Comment by MichaelClerx** [comment]
PSO has multiple particles flying through the parameter space, and each of those tracks its own local best. Keep in mind though, that it doesn't hone in on optima very carefully but rather flies around them quasi-randomly. This is the strength of PSO; it has lots of momentum on the particles to help them avoid local optima; but it's also a weakness, as it doesn't converge to the optima very well. There's also no guarantee whatsoever it finds them all (or even more than 1 of them). So if you're looking for a bunch of points that might be near some optima that's definitely something it can do
